content/blog/azure-pipelines-scaler.md

```diff
@@ -0,0 +1,224 @@
+              secretKeyRef:
+                name: azdevops
+                key: AZP_POOL
+        volumeMounts:
+        - mountPath: /var/run/docker.sock
+          name: docker-volume
+      volumes:
+      - name: docker-volume
+        hostPath:
+          path: /var/run/docker.sock
+```
+
+### Autoscaling with KEDA
+
+After the deployment is created you need to create the `ScaledObject` in order for KEDA to start scaling the deployment.
+To scale based on the queue length of an Azure Pipelines agent pool, you can use the `azure-pipelines` trigger.
+
+```yaml
+apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: pipeline-trigger-auth
+spec:
+  secretTargetRef:
+    - parameter: personalAccessToken
+      name: pipeline-auth
+      key: personalAccessToken
+---
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: azure-pipelines-scaledobject
+spec:
+  scaleTargetRef:
+    name: azdevops-deployment
+  minReplicaCount: 1
+  maxReplicaCount: 5 
+  triggers:
+  - type: azure-pipelines
+    metadata:
+      poolID: "1"
+      organizationURLFromEnv: "AZP_URL"
+    authenticationRef:
+     name: pipeline-trigger-auth
+```
+
+The default `targetPipelinesQueueLength` is `1`, so there will be one agent for each job.
+
+> The Azure Pipelines scaler supports scaling to zero but you need at least one agent registered in the agent pool in order for new jobs to be scheduled on the pool.
+
+### Running Azure Pipelines jobs
+
+After deploying the agent and the KEDA ScaledObject. It is time to see the autoscaling in action.
+
+First, check the current pods running in the deployment:
+```sh
+$ kubectl get pods
+NAME                                   READY   STATUS    RESTARTS   AGE
+azdevops-deployment-5854bbbf84-r86qv   1/1     Running   0          75s
+```
+
+![deployment agents](/img/blog/azure-pipelines-scaler/deployment-agents.png)
+
+Now let's queue some builds.
+
+![azure devops builds](/img/blog/azure-pipelines-scaler/deployment-builds.png)
+
+```sh
+$ kubectl get pods
+NAME                                   READY   STATUS    RESTARTS   AGE
+azdevops-deployment-5854bbbf84-4gfbx   1/1     Running   0          36s
+azdevops-deployment-5854bbbf84-r86qv   1/1     Running   0          12m
+azdevops-deployment-5854bbbf84-tm47k   1/1     Running   0          36s
+```
+
+![deployment agents](/img/blog/azure-pipelines-scaler/deployment-agents-autoscaled.png)
+
+## Run the self-hosted agent as a Job
+
+When running your agents as a deployment you have no control on which pod gets killed when scaling down. ([see KEDA docs](https://keda.sh/docs/1.4/concepts/scaling-deployments/#long-running-executions))
+
+If you run your agents as a `Job`, KEDA will start a Kubernetes job for each job that is in the agent pool queue. The agents will accept one job when they are started and terminate afterwards.
+You also achieve fully isolated build environments when using jobs, since an agent is always created for each job.
```

Comment:
How about this?

```suggestion
Since an agent is always created for every pipeline job, you can achieve fully isolated build environments by using Kubernetes jobs.
```